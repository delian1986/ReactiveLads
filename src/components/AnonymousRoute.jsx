import { Route, Redirect } from "react-router-dom";
import PropTypes from "prop-types";

function AnonymousRoute({ component: Component, isLoggedIn, ...rest }) {
  return (
    <Route
      {...rest}
      render={(props) => {
        if (isLoggedIn) {
          return (
            <Redirect to={{ pathname: "/", state: { from: props.location } }} />
          );
        }

        return <Component {...props} />;
      }}
    />
  );
}

export { AnonymousRoute };

AnonymousRoute.propTypes = {
  component: PropTypes.elementType,
  isLoggedIn: PropTypes.bool,
  rest: PropTypes.any
};
