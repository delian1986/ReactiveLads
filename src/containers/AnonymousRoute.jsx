import { getLoggedInStatus } from "../store/selectors";
import { AnonymousRoute } from "../components/AnonymousRoute";
import { connect } from "react-redux";

const mapStateToProps = (state) => {
  return {
    isLoggedIn: getLoggedInStatus(state)
  };
};

export default connect(mapStateToProps)(AnonymousRoute);
