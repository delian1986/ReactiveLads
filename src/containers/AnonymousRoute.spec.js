import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import AnonymousRoute from "./AnonymousRoute";
import { BrowserRouter } from "react-router-dom";

const mockStore = configureStore([]);

describe("tests <AnonymousRoute/> component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      auth: {
        isLoggedIn: false
      }
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <AnonymousRoute component={() => <></>} />
        </Provider>
      </BrowserRouter>
    );
  });
  describe("Test rendering component", () => {
    it("should render with given state from Redux store", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
