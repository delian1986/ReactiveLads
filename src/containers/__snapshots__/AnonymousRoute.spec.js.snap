// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests <AnonymousRoute/> component Test rendering component should render with given state from Redux store 1`] = `null`;
